### PR TITLE
Add Vulkan support

### PR DESCRIPTION
**Pull request by mlehtima**
Move common window handling code to shared location. Remove support for wayland < 1.15 to simplify code. Silence some build warnings.

### CODE REVIEW

File: hybris/egl/platforms/common/eglnativewindowbase.cpp [hunk 1]
```diff
@@ -0,0 +1,3 @@
+
+
+#include "eglnativewindowbase.h"
```
Comment (krnlyng) [inline hybris/egl/platforms/common/eglnativewindowbase.cpp:3]: Do we need this file for something?

File: hybris/egl/platforms/common/eglplatformcommon.cpp [hunk 2]
```diff
@@ -1,3 +1,20 @@
+/*
+ * Copyright (c) 2013-2019 Jolla Ltd.
```
Comment (krnlyng) [inline hybris/egl/platforms/common/eglplatformcommon.cpp:2]: 2022?

Comment (krnlyng) [inline hybris/egl/platforms/common/eglplatformcommon.cpp:2]: 2022?

File: hybris/egl/platforms/common/eglnativewindowbase.cpp (hunk 1 above)
Comment (mlehtima) [reply hybris/egl/platforms/common/eglnativewindowbase.cpp]: No, removed the useless file.

File: hybris/egl/platforms/common/eglplatformcommon.cpp (hunk 2 above)
Comment (mlehtima) [reply hybris/egl/platforms/common/eglplatformcommon.cpp]: Fixed

Comment (mlehtima) [reply hybris/egl/platforms/common/eglplatformcommon.cpp]: Fixed

File: hybris/vulkan/vulkan.c [hunk 3]
```diff
@@ -104,6 +105,41 @@ VkResult vkEnumerateInstanceExtensionProperties(const char* pLayerName, uint32_t
...
+{
+    VkResult ret;
+    if (_vkGetPhysicalDeviceSurfaceCapabilitiesKHR == NULL) {
+        _vkGetPhysicalDeviceSurfaceCapabilitiesKHR = (VkResult (*)(VkPhysicalDevice, VkSurfaceKHR, VkSurfaceCapabilitiesKHR*))
+            (*_vkGetInstanceProcAddr)((VkInstance)physicalDevice, "vkGetPhysicalDeviceSurfaceCapabilitiesKHR");
+    }
+
+    ret = (*_vkGetPhysicalDeviceSurfaceCapabilitiesKHR)(physicalDevice, surface, pSurfaceCapabilities);
+    // Surface size will be determined by the extent of a swapchain targeting the surface
+    pSurfaceCapabilities->currentExtent = (VkExtent2D){0xFFFFFFFF, 0xFFFFFFFF};
```
Comment (krnlyng) [inline hybris/vulkan/vulkan.c:118]: this looks a bit weird, can you make the comment a bit more explanatory?

Comment (mlehtima) [reply hybris/vulkan/vulkan.c]: That was an old remnant from some early testing, the code was not needed based on more testing.

**Review by krnlyng — APPROVED** [review]

**Pull request merged**